gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/GqlQuery.java

```diff
@@ -400,23 +387,27 @@ protected void populatePb(DatastoreV1.RunQueryRequest.Builder requestPb) {
   @Override
   protected Object fromPb(ResultType<V> resultType, String namespace, byte[] bytesPb)
       throws InvalidProtocolBufferException {
-    return fromPb(resultType, namespace, DatastoreV1.GqlQuery.parseFrom(bytesPb));
+    return fromPb(resultType, namespace, com.google.datastore.v1beta3.GqlQuery.parseFrom(bytesPb));
   }
 
   private static <V> GqlQuery<V> fromPb(
-      ResultType<V> resultType, String ns, DatastoreV1.GqlQuery queryPb) {
+      ResultType<V> resultType, String ns, com.google.datastore.v1beta3.GqlQuery queryPb) {
     Builder<V> builder = new Builder<>(resultType, queryPb.getQueryString());
     builder.namespace(ns);
-    if (queryPb.hasAllowLiteral()) {
-      builder.allowLiteral = queryPb.getAllowLiteral();
-    }
-    for (DatastoreV1.GqlQueryArg nameArg : queryPb.getNameArgList()) {
-      Binding argument = Binding.fromPb(nameArg);
-      builder.namedBindings.put(argument.name(), argument);
+    builder.allowLiteral = queryPb.getAllowLiterals();
+    for (Map.Entry<String, com.google.datastore.v1beta3.GqlQueryParameter> nameArg
+         : queryPb.getNamedBindings().entrySet()) {
+      Binding currBinding = Binding.fromPb(nameArg.getValue());
+      if (currBinding != null) {
```

Comment:
As discussed, I don't think we need this check (or the one bellow).
If you want you get add a `Precondition.checkState` in `Binding.fromPb`